pkg/azuredisk/nodeserver_v2.go

```diff
@@ -90,7 +93,32 @@ func (d *DriverV2) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolume
 		return nil, status.Errorf(codes.Internal, "Failed to find disk on lun %s. %v", lunStr, err)
 	}
 
-	source, err := d.nodeProvisioner.GetDevicePathWithLUN(int(lun))
+	// if mount recovery is necessary, detach volume so that volume can be reattached via reconciliation
+	if d.nodeProvisioner.MountRecoveryRequired(int(lun)) {
```

Comment:
If I follow the logic correctly, the first two `NodeStageVolume` calls will fail before we attempt to reconcile on the third call. Shouldn't start the process after the first failure?

Reply:
Your observation is correct. The intention is to wait for the `GetDevicePathWithLun` to fail at least twice (adding up to at least 4 minutes of iscsi scan) before initiating the recovery sequence.